### DISCUSSION TITLE
Should I do HASPone or openHASP?

### DISCUSSION THREAD

**Discussion by aph3rson** [HASwitchPlate/HASPone#235] [General]
(I couldn't find a good place to ask a general HASwitchplate question, so here goes.)

I'm looking at deploying a few new touchscreens into my home. Here's the important things to know about the deployment:

- Switches are going into multi-gang North American junction boxes (2/3/4-gang), where other switches will be residing.
- The touchscreens shouldn't need to control another dimmer/switch/etc., though it might be nice to have for the future.
- I'm comfortable with self-assembly and electrical work, though the j-boxes themselves cannot be replaced.
- I have access to 3D printing hardware, but not skills to make my own 3D models for customized prints.
- I can/have ordered PCBs before from short-run printers, but I don't have skills for making my own PCBs (could learn, though).

Given that, I think I have three different options:

| Hardware                    | Pros                                                                                                                                       | Cons                                                                                                                                                                                     |
| --------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| HASPone                     | <ul><li>Relatively mature devices</li><li>BOM/parts easily available</li><li>Shell/faceplate STLs available</li><li>Easy to configure in HA</li></ul> | <ul><li>All of Nextion's tomfoolery</li><li>Limited number of pages</li><li>No video support possible</li></ul> |
| openHASP<br>(Lanbon L8)     | <ul><li>Readily-available devices</li><li>Many used w/ openHASP</li><li>In-wall wiring control</li><li>Better display performance</li></ul> | <ul><li>Existing faceplate will not fit multi-gang box<br>(would need to make/print custom shell?)</li><li>Weird mounting configuration</li><li>More difficult to program/configure vs. HASPone</li></ul> |
| openHASP<br>(HASPone shell) | <ul><li>Best display performance</li><li>Shell/faceplate STLs available</li><li>Most customizable</li></ul> | <ul><li>Might need to make my own PCB?</li><li>BOM would need to be customized</li><li>Most difficult to program/configure</li><li>🐙 _Uncharted waters_ 🐙</li></ul> |

Did I get that all right? What might you all recommend out of the three options?

**Comment by aderusha** [comment]
My biggest issue with the market at large is that, outside of the HASPone, none of the existing options can fit into a multi-gang workbox in North America.  It's the sole reason I developed HASPone and am a little surprised that there are no commercial options yet.  EU style boxes are well covered by an increasing number of solutions today, including the Lanbon and first-party Nextion offerings.

So, if you're in NA and you want a multi-gang setup, I think we're kinda stuck w/ HASPone.  Obviously, my opinion here is biased toward the solution which I am most familiar with :D

**Reply by aph3rson** [reply]
@aderusha thanks for the insight there. How possible would you think it'd be for openHASP hardware to live inside of a HASPone shell? 

As far as I know, this mostly means using a different screen, and getting SPI to the display shield instead of UART.  Everything else looks? like it'd be board-compatible.

**Comment by aderusha** [comment]
I don't know that much of the existing design work would carry over at all, you'd be started with a new PCB design and new enclosure models to fit the new components.  It'd certainly be possible but there's not much that'd be directly re-usable from the haspONE project.

**Comment by fvanroie** [comment]
I think there is no need to design a new PCB, since there are several dozens of ESP32(S3) boards with integrated display (2.4" to 7").
You still need a 5V power supply, but perhaps the HASPone PCB can be repurposed solely as a 5V power supply.

Depending on the amount of work you want to do, you can try to:
- Adjust the faceplate design to fit either the 3.5" [WT32-SC01-Plus](https://www.openhasp.com/0.7.0/hardware/wireless-tag/wt32-sc01-plus/#features) or the 4" [Panlee ZW3D95CE01S-AR-4848 8](https://community.home-assistant.io/t/4-wall-switch-with-esp32-captive-touch-480x480-under-50-us-with-shipping/559496) _aka_ [WT32S3-86V](https://github.com/HASwitchPlate/openHASP/discussions/435).
- Try the [Lolin 2.4" TFT](https://www.openhasp.com/0.7.0/hardware/lolin/lolin-tft-shield/) which has roughly has the same dimensions as a Nextion 2.4" (not yet tested but looks compatible with HASPone)

With access to a 3D printer, I think you should look into 3D CAD design over learning PCB design... at least for this project.
Or just go with the provided designs and the HASPone.

I'm not from the US, so I'm not familiar with the wall-boxes used there. Going by @aderusha's 3D designs, I believe these could be adjusted to fit a 3.5" or 4" ESP32 board too... or try your luck with the 2.4" Lolin display.

**Reply by fvanroie** [reply]
Like this:
![image](https://github.com/HASwitchPlate/HASPone/assets/15969459/9aab2735-8896-425d-beef-ac64dbbbaf1f)

**Reply by stampeder** [reply]
Nice to see someone that aligns the screws. Shows pride in their work....Looks really nice.
